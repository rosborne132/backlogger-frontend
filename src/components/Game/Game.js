import React, { PureComponent } from "react"
import PropTypes from "prop-types"
import { Link } from "react-router-dom"
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faMinusSquare, faPenSquare } from '@fortawesome/free-solid-svg-icons'

import { GameIcons, GameStyles, GameHeader, GameBody } from '../StyledComponents'
// import config from "../config"

import GamesContext from '../../context/GamesContext'

class Game extends PureComponent {

  static contextType = GamesContext

  handleDeleteGame = e => {
    e.preventDefault()
    const { id, name } = this.props
    console.log(`${name}: ${id}`)
    this.context.deleteGame(id)
    // fetch(`${config.NOTE_API_ENDPOINT}/${noteId}`, {
    //   method: "DELETE",
    //   headers: {
    //     "content-type": "application/json"
    //   }
    // })
    //   .then(res => {
    //     if (!res.ok) return res.json().then(e => Promise.reject(e))
    //     return res.json()
    //   })
    //   .then(() => {
    //     this.context.deleteNote(noteId)
    //   })
    //   .catch(error => {
    //     console.error({ error })
    //   })
  }

  render() {
    const { id, name } = this.props
    return (
      <GameStyles>
        <GameHeader>
          <Link to={`/app/game/${id}`}>{name}</Link>
        </GameHeader>

        <GameBody></GameBody>

        <GameIcons>
            <Link to={`#`} onClick={this.handleDeleteNote}><FontAwesomeIcon onClick={this.handleDeleteGame} icon={faMinusSquare} /></Link>
            <Link to={`/app/updateGame/${id}`}><FontAwesomeIcon icon={faPenSquare} /></Link>
        </GameIcons>
        
        
        
      </GameStyles>
    )
  }
}

Game.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
}

Game.defaultProps = {
  onDeleteGame: () => {}
}

export default Game
