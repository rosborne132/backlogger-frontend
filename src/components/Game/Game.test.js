import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import Game from './Game'

it('renders empty given no tabs', () => {
    const component = shallow(<Game id="20394" name="test"/>)
    expect(component).toMatchSnapshot()
})