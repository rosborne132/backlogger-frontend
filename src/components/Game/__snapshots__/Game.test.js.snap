// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty given no tabs 1`] = `ShallowWrapper {}`;
